htdocs/user/class/user.class.php

```diff
@@ -3214,7 +3214,7 @@ function fetchAll($sortorder='', $sortfield='', $limit=0, $offset=0, $filter=arr
 
 		$sql="SELECT t.rowid";
 		$sql.= ' FROM '.MAIN_DB_PREFIX .$this->table_element.' as t ';
-		$sql.= " WHERE 1";
+		$sql.= " WHERE t.entity IN (".getEntity('user').")";
```

Comment:
@ptibogxiv 
maybe send this in "$filter" parameter when you need it?

Reply:
@hregis i want  in no transverse mode, only display users of the entity in email form

Reply:
@ptibogxiv 
this breaks the function "fetchAll" as the name suggests! In addition it is not compatible with the "transversal" mode.

Reply:
@ptibogxiv @eldy 
can you try with this patch? : https://github.com/Dolibarr/dolibarr/pull/11060

Reply:
#11060 is good for me, better select of user by entity